compactor/src/utils.rs

```diff
@@ -199,17 +200,33 @@ pub(crate) fn compute_split_time(
     let mut min = min_time;
     loop {
         let split_time = min + ((max_time - min_time) as f64 * percentage).ceil() as i64;
-        if split_time < max_time {
-            split_times.push(split_time);
-            min = split_time;
-        } else {
+
+        if split_time >= max_time {
             break;
+        } else if time_range_present(chunk_times.clone(), min, split_time) {
+            split_times.push(split_time);
         }
+        min = split_time;
     }
 
     split_times
 }
 
+// time_range_present returns true if the given time range is included in any of the chunks.
+fn time_range_present(
+    chunk_times: Vec<(i64, i64)>, // Vec of (min_time, max_time)
+    min_time: i64,
+    max_time: i64,
+) -> bool {
+    for chunk in chunk_times {
+        // if chunk.max >= min_time and chunk.min <= max_time, this chunk is within the given time range.
+        if chunk.1 >= min_time && chunk.0 <= max_time {
+            return true;
+        }
+    }
+    false
+}
```

Comment:
So this function can borrow a slice of `chunk_times` to eliminate the `clone` above, and then [`Iterator::any`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.any) does exactly what you've implemented here:

```suggestion
fn time_range_present(
    chunk_times: &[(i64, i64)],
    min_time: i64,
    max_time: i64,
) -> bool {
    chunk_times.iter().any(|&(chunk_min, chunk_max)| {
        chunk_max >= min_time && chunk_min <= max_time
    })
}
```

Again, please let me know if this doesn't make sense! I highly recommend to folks new to Rust to skim through the `Iterator` methods so that you have an idea of all the capabilities it provides for you :)